.github/workflows/test_cocoapods_integration.yml

```diff
@@ -12,24 +12,27 @@ on:
 jobs:
 
   pods:
-    runs-on: macos-12
+    runs-on: macos-14-xlarge
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - uses: n1hility/cancel-previous-runs@v3
       with:
         token: ${{ secrets.GITHUB_TOKEN }}
 
     - name: Select latest Xcode
       uses: maxim-lobanov/setup-xcode@v1
       with:
-        xcode-version: latest-stable
+        xcode-version: '16.1'
```

Comment:
This looks repetitive, I guess it could be extracted to the top level as env var
```
env:
  xcode_version: '16.1'
```

and later used as 
```
with:
    xcode-version: ${{env.xcode_version}}
````